src/wallet/signer.rs

```diff
@@ -401,26 +410,36 @@ impl InputSigner for SignerWrapper<PrivateKey> {
             if let Some((leaf_hashes, _)) =
                 psbt.inputs[input_index].tap_key_origins.get(&x_only_pubkey)
             {
-                let leaf_hashes = leaf_hashes
-                    .iter()
-                    .filter(|lh| {
-                        !psbt.inputs[input_index]
-                            .tap_script_sigs
-                            .contains_key(&(x_only_pubkey, **lh))
-                    })
-                    .cloned()
-                    .collect::<Vec<_>>();
-                for lh in leaf_hashes {
-                    let (hash, hash_ty) = Tap::sighash(psbt, input_index, Some(lh))?;
-                    sign_psbt_schnorr(
-                        &self.inner,
-                        x_only_pubkey,
-                        Some(lh),
-                        &mut psbt.inputs[input_index],
-                        hash,
-                        hash_ty,
-                        secp,
-                    );
+                if sign_options.tap_leaves_options != TapLeavesOptions::None {
```

Comment:
I would remove this if and move it into the match. `None` always returns false, `All` always true